src/main/java/org/broadinstitute/consent/http/service/DatabaseElectionAPI.java

```diff
@@ -628,13 +632,15 @@ private void validateAvailableUsers(Election election) {
         }
     }
 
-    private void updateSortDate(String referenceId, Date createDate){
-        if(consentDAO.checkConsentById(referenceId) != null){
+    private void updateSortDate(String referenceId, Date createDate) {
+        if (Objects.nonNull(consentDAO.checkConsentById(referenceId))) {
             consentDAO.updateConsentSortDate(referenceId, createDate);
         } else {
-            Document dar = dataAccessRequestService.getDataAccessRequestByReferenceIdAsDocument(referenceId);
-            dar.put(DarConstants.SORT_DATE, createDate.getTime());
-            dataAccessRequestService.updateDocumentByReferenceId(referenceId, dar);
+            DataAccessRequest dar = dataAccessRequestService.findByReferenceId(referenceId);
+            dar.setSortDate(new Timestamp(createDate.getTime()));
+            dar.getData().setSortDate(createDate.getTime());
+            User user = userDAO.findUserById(dar.getUserId());
+            dataAccessRequestService.updateByReferenceIdVersion2(user, dar);
         }
     }
```

Comment:
This is essentially a `Document` -> `DataAccessRequest` refactor which also solved the primary bug of throwing a NFE